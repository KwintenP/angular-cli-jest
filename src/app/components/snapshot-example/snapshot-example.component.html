<p>
  snapshot-example workss!
</p>
{{title}}
<!--test-->

<div *ngFor="let datum of data">
  {{datum}}
</div>

<div *ngIf="visible">
  Dit is zichtbaar
</div>
