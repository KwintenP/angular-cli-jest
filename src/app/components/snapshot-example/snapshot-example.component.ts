import {Component, OnInit} from '@angular/core';
import {Observable} from 'rxjs/Observable';
import 'rxjs/add/observable/interval';

@Component({
  selector: 'app-snapshot-example',
  templateUrl: './snapshot-example.component.html',
  styleUrls: ['./snapshot-example.component.css']
})
export class SnapshotExampleComponent implements OnInit {
  title: string;
  data: [any];

  visible = false;

  constructor() {
  }

  ngOnInit() {
    this.title = 'Dit is een dynamisch titeltje';
    this.data = ['test', 'test2', 'test3'];
  }

}
