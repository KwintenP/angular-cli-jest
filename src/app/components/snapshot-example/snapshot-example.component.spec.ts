import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { SnapshotExampleComponent } from './snapshot-example.component';

describe('SnapshotExampleComponent', () => {
  let component: SnapshotExampleComponent;
  let fixture: ComponentFixture<SnapshotExampleComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ SnapshotExampleComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SnapshotExampleComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should match the previous snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });


  it('should match the previous snapshot2', () => {
    component.visible = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
