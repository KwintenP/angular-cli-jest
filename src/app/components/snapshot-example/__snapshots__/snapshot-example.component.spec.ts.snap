// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotExampleComponent should match the previous snapshot 1`] = `
<app-snapshot-example
  data={[Function Array]}
  title={[Function String]}
  visible={[Function Boolean]}
>
  <p>
    snapshot-example workss!
  </p>Dit is een dynamisch titeltje<div>
    test
  </div><div>
    test2
  </div><div>
    test3
  </div><div>
    Dit is zichtbaar
  </div>
</app-snapshot-example>
`;

exports[`SnapshotExampleComponent should match the previous snapshot2 1`] = `
<app-snapshot-example
  data={[Function Array]}
  title={[Function String]}
  visible={[Function Boolean]}
>
  <p>
    snapshot-example workss!
  </p>Dit is een dynamisch titeltje<div>
    test
  </div><div>
    test2
  </div><div>
    test3
  </div><div>
    Dit is zichtbaar
  </div>
</app-snapshot-example>
`;
